### PR TITLE
Update AB/BSE COS Requirements through F2019

### PR DESCRIPTION
**Pull request by Daniel-Wey**
Updated course requirements for both AB COS and BSE COS majors.
Updated contact information for undergrad faculty representatives for both AB and BSE COS.

### CODE REVIEW

File: tigerpath/majors_and_certificates/majors/COS-AB_2018.json [hunk 1]
```diff
@@ -212,7 +214,7 @@
...
+          "explanation": "<i>Editor's Note: Beginning with the Class of 2023 CHM 301, ECO 312 and Physics courses will no longer count as COS departmentals.</i>\n\nAny computer science (COS) course at the 300 or 400 level (except independent work) counts as a computer science departmental.\nIn addition, any 300- or 400- level Math or ELE or Physics or ORF course that does not duplicate COS content, Philosophy 312, MAE 345, CHM 303(CHM 303 is now renumbered to CHM 301 as of 8/27/2018), ECO 312, MOL 437/NEU 437, NEU 330, MUS 315, and ECO 326 – Economics of the Internet count as departmentals.  On occasion, special courses with computational content from other departments may fall in to this category.",
```
Comment (bnehoran) [inline tigerpath/majors_and_certificates/majors/COS-AB_2018.json]:
```suggestion
          "explanation": "<i>Note: Beginning with the Class of 2023 CHM 301, ECO 312 and Physics courses will no longer count as COS departmentals.</i>\n\nAny computer science (COS) course at the 300 or 400 level (except independent work) counts as a computer science departmental.\nIn addition, any 300- or 400- level Math or ELE or Physics or ORF course that does not duplicate COS content, Philosophy 312, MAE 345, CHM 303(CHM 303 is now renumbered to CHM 301 as of 8/27/2018), ECO 312, MOL 437/NEU 437, NEU 330, MUS 315, and ECO 326 – Economics of the Internet count as departmentals.  On occasion, special courses with computational content from other departments may fall in to this category.",
```
Editor's note means more that you, the editor, are adding the note. In this case it comes from directly from the department.

File: tigerpath/majors_and_certificates/majors/COS-BSE_2018.json [hunk 2]
```diff
@@ -158,17 +159,18 @@
                 "COS 314/MUS 314: Computer and Electronic Music through Programming, Performance, and Composition",
                 "COS 323/ORF 363: Computing and Optimization for the Physical and Social Sciences",
                 "COS 324: Introduction to Machine Learning",
-                "COS 325/MUS 315: Transforming Reality by Computer",
                 "COS 326: Functional Programming",
                 "COS 343: Algorithms for Computational Biology",
-                "COS 401/TRA 301: Introduction to Machine Translation ",
+                "COS 360/PSY 360: Computational Models of Cognition",
+                "COS 401/TRA 301: Introduction to Machine Translation",
```
Comment (bnehoran) [inline tigerpath/majors_and_certificates/majors/COS-BSE_2018.json:166]: Good catch! (the extra space)
In reality the names here are just for reference (that is, everything after the colon is ignored), but good to keep it neat and consistent.

File: tigerpath/majors_and_certificates/majors/COS-AB_2018.json [hunk 3]
```diff
@@ -182,9 +183,9 @@
                 "COS 314/MUS 314: Computer and Electronic Music through Programming, Performance, and Composition",
                 "COS 323/ORF 363: Computing and Optimization for the Physical and Social Sciences",
                 "COS 324: Introduction to Machine Learning",
-                "COS 325/MUS 315: Transforming Reality by Computer",
                 "COS 326: Functional Programming",
                 "COS 343: Algorithms for Computational Biology",
+                "COS 360/PSY 360: Computational Models of Cognition",
                 "COS 401/TRA 301: Introduction to Machine Translation ",
```
Comment (bnehoran) [inline tigerpath/majors_and_certificates/majors/COS-AB_2018.json]:
```suggestion
                 "COS 401/TRA 301: Introduction to Machine Translation",
```
Just, if we're already fixing it in one, then we might as well fix it in the other.

File: tigerpath/majors_and_certificates/majors/COS-BSE_2018.json [hunk 4]
```diff
@@ -188,7 +190,7 @@
...
+          "explanation": "<i>Editor's Note: Beginning with the Class of 2023 CHM 301, ECO 312 and Physics courses will no longer count as COS departmentals.</i>\n\nAny computer science (COS) course at the 300 or 400 level (except independent work) counts as a computer science departmental.\nIn addition, any 300- or 400- level Math or ELE or Physics or ORF course that does not duplicate COS content, Philosophy 312, MAE 345, CHM 303(CHM 303 is now renumbered to CHM 301 as of 8/27/2018), ECO 312, MOL 437/NEU 437, NEU 330, MUS 315, and ECO 326 – Economics of the Internet count as departmentals.  On occasion, special courses with computational content from other departments may fall in to this category.",
```
Comment (bnehoran) [inline tigerpath/majors_and_certificates/majors/COS-BSE_2018.json]:
```suggestion
          "explanation": "<i>Note: Beginning with the Class of 2023 CHM 301, ECO 312 and Physics courses will no longer count as COS departmentals.</i>\n\nAny computer science (COS) course at the 300 or 400 level (except independent work) counts as a computer science departmental.\nIn addition, any 300- or 400- level Math or ELE or Physics or ORF course that does not duplicate COS content, Philosophy 312, MAE 345, CHM 303(CHM 303 is now renumbered to CHM 301 as of 8/27/2018), ECO 312, MOL 437/NEU 437, NEU 330, MUS 315, and ECO 326 – Economics of the Internet count as departmentals.  On occasion, special courses with computational content from other departments may fall in to this category.",
```
Same as in the AB version.

**Review by bnehoran — CHANGES_REQUESTED** [review]
Looking good! You can test out your changes on heroku by pressing the "View Deployment" button above or at the bottom under "Show Environments".

See my individual comments for specific feedback, but one additional thing: I think COS 511 is missing from Theory, isn't it?

**Comment by bnehoran** [comment]
Oh, also this is not too important, but it's customary to add a short description to the pull request summarizing the changes and/or why they were made.
You can also look at previous pull requests/commits for how to format the title of the PR.
These are just nits, but they help a lot when looking back through the history.

**Comment by Daniel-Wey** [comment]
> See my individual comments for specific feedback, but one additional thing: I think COS 511 is missing from Theory, isn't it?

Done

**Comment by Daniel-Wey** [comment]
> Oh, also this is not too important, but it's customary to add a short description to the pull request summarizing the changes and/or why they were made.
> You can also look at previous pull requests/commits for how to format the title of the PR.
> These are just nits, but they help a lot when looking back through the history.

Done

File: tigerpath/majors_and_certificates/majors/COS-BSE_2018.json (hunk 2 above)
Comment (Daniel-Wey) [reply tigerpath/majors_and_certificates/majors/COS-BSE_2018.json:166]: Understood!

**Review by bnehoran — APPROVED** [review]

**Pull request merged**